### PR TITLE
Introduce ExtendedSQLTest tag to split build jobs

### PR DESCRIPTION
**Pull request by turboFei**
### _Why are the changes needed?_

Involve ExtendedSQLTest tag to split build jobs

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by turboFei** [comment]
maybe we can remove some jobs.

For ExtendedSQLTest, just keep:
- spark-3.0
- spark-3.1
- spark-3.2.0-SNAPSHOT

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -33,11 +33,14 @@ jobs:
     strategy:
       matrix:
         profiles:
-          - ''
-          - '-Pspark-3.0 -Dspark.archive.mirror=https://archive.apache.org/dist/spark/spark-3.1.1 -Dspark.archive.name=spark-3.1.1-bin-hadoop2.7.tgz -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.DataLakeTest'
-          - '-Pspark-3.1'
-          - '-Pspark-3.1 -Pspark-hadoop-3.2'
-          - '-Pspark-3.2-snapshot -pl :kyuubi-spark-sql-engine,:kyuubi-common,:kyuubi-ha,:kyuubi-zookeeper -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.DataLakeTest'
+          - '-Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.ExtendedSQLTest'
```
Comment (yaooqinn) [inline .github/workflows/master.yml:36]: How about making it default in pom?

**Comment by yaooqinn** [comment]
maybe enabling 3.1 and 3.2 is enough?

**Comment by ulysses-you** [comment]
Can we move these tag to `pom.xml` is possible ? not only for GA but also travis.

**Comment by turboFei** [comment]
> Can we move these tag to `pom.xml` if possible ? not only for GA but also travis.

move  EXCLUDE_TAGS/INCLUDE_TAGS into PROFILE?

cc @yaooqinn

**Comment by yaooqinn** [comment]
I am ok with the current setting, but let's remove some duplicated workflows

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/datalake/IcebergOperationSuite.scala [hunk 2]
```diff
@@ -19,8 +19,9 @@ package org.apache.kyuubi.operation.datalake
 
 import org.apache.kyuubi.config.KyuubiConf
 import org.apache.kyuubi.operation.{BasicIcebergJDBCTests, WithKyuubiServer}
-import org.apache.kyuubi.tags.DataLakeTest
+import org.apache.kyuubi.tags.{DataLakeTest, ExtendedSQLTest}
 
+@ExtendedSQLTest
```
Comment (yaooqinn) [inline kyuubi-main/src/test/scala/org/apache/kyuubi/operation/datalake/IcebergOperationSuite.scala:24]: we can remove this tag to make the latter usage simple?

**Comment by yaooqinn** [comment]
Let's mark the test suites with one tag only to make the caller side simpler

For now, `ExtendedSQLTest` should be marked to TPCDS tests, and defined as `exclude` for all workflows in pom directly. So I guess the current workflow stays the same as it is now.

Add one or two separate jobs to test the TPCDS test cases only, SHOULD NOT use the tag `ExtendedSQLTest` here

**Comment by yaooqinn** [comment]
THAT said, the newly added jobs can simply do something like `build/mvn -Dtest=none -DwildcardSuites=the test name test`

File: kyuubi-main/src/test/scala/org/apache/kyuubi/operation/datalake/IcebergOperationSuite.scala (hunk 2 above)
Comment (turboFei) [reply kyuubi-main/src/test/scala/org/apache/kyuubi/operation/datalake/IcebergOperationSuite.scala:24]: ok

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**